### DISCUSSION TITLE
AsyncLoadingCache with reactive cache loading

### DISCUSSION THREAD

**Discussion by jeromenicholas07** [ben-manes/caffeine#1524] [Q&A]
Can anybody please confirm if the following is the correct way to initialize AsyncLoadingCache?
I checked the other discussion threads about using an `AsyncCache` to prevent `cache stampede` in a reactive application, Would an `AsyncLoadingCache` achieve the same results if initialized in this manner:
```
@Component
public class Retriever  {
      private AsyncLoadingCache<String, String> CACHE;

      @PostConstruct
      public void initialize() {
          CACHE = Caffeine.newBuilder()
                  .expireAfterWrite(Duration.ofMinutes(5))
                  .maximumSize(1000)
                  .buildAsync((key, executor) -> Mono.defer(() -> networkCall(key))
                  .subscribeOn(Schedulers.fromExecutor(executor)).toFuture());
      }

      private Mono<String> networkCall(String key) {...}

    //      Usage
    CACHE.get(identifier)
}
```
Are there any caveats to this approach that I should be wary of?

**Comment by ben-manes** [comment]
It looks reasonable to me, but I haven’t used reactor very much
